microcloud/test/suites/basic.sh

```diff
@@ -822,3 +822,135 @@ test_auto() {
     ! lxc exec ${m} -- lxc network ls -f csv | grep -q "^UPLINK," || false
   done
 }
+
+# services_validator: A basic validator of 3 systems with typical expected inputs.
+services_validator() {
+  for m in micro01 micro02 micro03 ; do
+    validate_system_lxd ${m} 3 disk1 1 1 enp6s0 10.1.123.1/24 10.1.123.100-10.1.123.254 fd42:1:1234:1234::1/64 10.1.123.1,8.8.8.8
+    validate_system_microceph ${m} 1 disk2
+    validate_system_microovn ${m}
+  done
+}
+
+test_reuse_cluster() {
+  unset_interactive_vars
+
+  # Set the default config for interactive setup.
+  export LOOKUP_IFACE="enp5s0"
+  export LIMIT_SUBNET="yes"
+  export EXPECT_PEERS=2
+  export SETUP_ZFS="yes"
+  export ZFS_FILTER="lxd_disk1"
+  export ZFS_WIPE="yes"
+  export SETUP_CEPH="yes"
+  export SETUP_CEPHFS="yes"
+  export CEPH_FILTER="lxd_disk2"
+  export CEPH_WIPE="yes"
+  export SETUP_OVN="yes"
+  export OVN_FILTER="enp6s0"
+  export IPV4_SUBNET="10.1.123.1/24"
+  export IPV4_START="10.1.123.100"
+  export IPV4_END="10.1.123.254"
+  export DNS_ADDRESSES="10.1.123.1,8.8.8.8"
+  export IPV6_SUBNET="fd42:1:1234:1234::1/64"
+
+  reset_systems 3 3 3
+  echo Create a MicroCloud that re-uses an existing service
+  export REUSE_EXISTING_COUNT=1
+  export REUSE_EXISTING="add"
+  lxc exec micro02 -- microceph cluster bootstrap
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+  services_validator
+
+  reset_systems 3 3 3
+  echo Create a MicroCloud that re-uses an existing service on the local node
+  lxc exec micro01 -- microceph cluster bootstrap
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+  services_validator
+
+  reset_systems 3 3 3
+  echo Create a MicroCloud that re-uses an existing MicroCeph and MicroOVN
+  export REUSE_EXISTING_COUNT=2
+  export REUSE_EXISTING="add"
+  lxc exec micro02 -- microceph cluster bootstrap
+  lxc exec micro02 -- microovn cluster bootstrap
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+  services_validator
+
+  reset_systems 3 3 3
+  echo Create a MicroCloud that re-uses an existing MicroCeph and MicroOVN on different nodes
+  lxc exec micro02 -- microceph cluster bootstrap
+  lxc exec micro03 -- microovn cluster bootstrap
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+  services_validator
+
+  reset_systems 3 3 3
+  echo Create a MicroCloud that re-uses an existing existing service with multiple nodes from this cluster
+  export REUSE_EXISTING_COUNT=1
+  export REUSE_EXISTING="add"
+  lxc exec micro02 -- microceph cluster bootstrap
+  token="$(lxc exec micro02 -- microceph cluster add micro01)"
```

Comment:
I didn't know`microceph` diverged in behavior from LXD where we need to pass `--quiet` to avoid showing the text header. Is that the case for `micro*`?

Reply:
the `micro*` projects don't have any header, they just spit out the token.